### ISSUE TITLE
Unable to see Magisk module updates

### ISSUE THREAD

**Issue by aravindvnair99** [topjohnwu/Magisk#3337]
In previous versions of Magisk Manager, we could easily make out which modules are outdated and require an update. But in the latest version, it doesn't display update prompt or in the list. One must manually search for the installed module name and check if the version number has been incremented. It would be nice to have the app list updates first or have a separate page for updates.

**Comment by zgfg** [comment]
If you have a module installed, and its newer version was updated to the Repo, you will see that at the top of the Modules tab/page

Currently I don't see any update (there is no one for my installed modules) but last time I saw it maybe with Canary Manager 307 when MHCP module was updated and it worked fine

Actually, being already for long on this 'new' design Canary (it started back in the spring) and updated alteady many modules that way, with no problem 

Do you have a particular module you have it installed, and you know that it was updated on Repo, and you don't see it with Stable/Beta 8.0.2 or Canary 310 at the top of the Modules page?

**Comment by aravindvnair99** [comment]
> If you have a module installed and its newer version was updated to the Repo, you will see the available update at the top of the Modules tab/page
> 
> Currently I don't see any update (there is no one for my installed modules) but last time I saw it maybe with Canary Manager 307 when MHCP module was updated and it worked fine
> 
> Actually, I'm already for long on this 'new' Canary design (it started back in the spring) and I have already updated many modules that way, with no problem
> 
> Do you have a particular module installed for whom you know that was updated on Repo, and you don't see it the abailable update with Stable/Beta 8.0.2 or Canary 310 at the top of the Modules page?

@zgfg Thanks for your reply. The list shows the most recently uploaded first. It's in chronological order. You probably are a frequent user of Magisk Manager and might not have noticed that, but let's say someone doesn't open it that often... He or she will definitely miss the update. Instead, in my opinion, it would be better to have a page specifically for updates or show the updates at the top of the list and then the remaining. It was the same in previous versions if I remember correctly, but it was slightly more evident. By showing it at the very top, we can avoid the hassle of scrolling through or hunting manually.

**Comment by aravindvnair99** [comment]
> If you have a module installed and its newer version was updated to the Repo, you will see the available update at the top of the Modules tab/page
> 
> Currently I don't see any update (there is no one for my installed modules) but last time I saw it maybe with Canary Manager 307 when MHCP module was updated and it worked fine
> 
> Actually, I'm already for long on this 'new' Canary design (it started back in the spring) and I have already updated many modules that way, with no problem
> 
> Do you have a particular module installed for whom you know that was updated on Repo, and you don't see it the abailable update with Stable/Beta 8.0.2 or Canary 310 at the top of the Modules page?

@zgfg Thanks for your reply. The list shows the most recently uploaded first. It's in chronological order. You probably are a frequent user of Magisk Manager and might not have noticed that, but let's say someone doesn't open it that often... He or she will definitely miss the update. Instead, in my opinion, it would be better to have a page specifically for updates or show the updates at the top of the list and then the remaining. It was the same in previous versions if I remember correctly, but it was slightly more evident. By showing it at the very top, we can avoid the hassle of scrolling through or hunting manually.

**Comment by Didgeridoohan** [comment]
@aravindvnair99 Updates do show at the top, if there is one...

If you are talking about the dates showing under the modules in the list, those do not necessarily mean that there has been an update to the module. That date just shows the last time there was a commit to the repo, which might just be a simple readme change and not something that requires an update of the module.

**Comment by aravindvnair99** [comment]
> @aravindvnair99 Updates do show at the top, if there is one...
> 
> If you are talking about the dates showing under the modules in the list, those do not necessarily mean that there has been an update to the module. That date just shows the last time there was a commit to the repo, which might just be a simple readme change and not something that requires an update of the module.

@Didgeridoohan Thanks for replying! Yes, I understand that. I was talking about the version number itself. Not dates. I'll install some outdated modules and post a few screenshots.

**Comment by aravindvnair99** [comment]
@zgfg @Didgeridoohan I just reinstalled all my modules all over again. Then installed an old version for a randomly selected module from my list.

I can confirm this isn't a valid issue and that I'm able to see the updates separately on top.

Not sure why it didn't work last time. I remember I manually checked the version and realised that I needed to upgrade for two of my installed modules.

Nevertheless, I've attached a screenshot showing that everything works fine and I'll close this issue. Thanks for clarifying! :)

![Screenshot_20201015-015859907.jpg](https://user-images.githubusercontent.com/22199259/96042735-7ddfe400-0e8b-11eb-920d-a9670b8207a2.jpg)

**Comment by Didgeridoohan** [comment]
No worries. Sometimes you need to refresh for the updates to show up (the little icon in the top right).

Great choice of module, by the way... 👍

**Comment by aravindvnair99** [comment]
> No worries. Sometimes you need to refresh for the updates to show up (the little icon in the top right).

Thanks, @Didgeridoohan! Will try that next time.

> Great choice of module, by the way... 👍

Wasn't intentional, I honestly realised only after I commented. But thanks a lot for your work! It's been a great saver for me! 😊

**Comment by aravindvnair99** [comment]
> @aravindvnair99 Updates do show at the top, if there is one...
> 
> If you are talking about the dates showing under the modules in the list, those do not necessarily mean that there has been an update to the module. That date just shows the last time there was a commit to the repo, which might just be a simple readme change and not something that requires an update of the module.

@Didgeridoohan I just noticed this... This is what you meant here right?

![Screenshot_20201112-041142931](https://user-images.githubusercontent.com/22199259/98873194-ee375080-249d-11eb-8061-05c32727326c.jpg)

**Comment by Didgeridoohan** [comment]
@aravindvnair99 There's nothing wrong there. The top one is what you have installed, the bottom one is the one in the modules repo.

The version difference is because the current fingerprints list is at v108, while you have v106 installed. The prints list is updated from the module (by running the props command), not from the Manager. If there had been an update to the module, the top one would have been under the "Updates" section (as you saw previously) instead of "Installed".

**Comment by aravindvnair99** [comment]
> @aravindvnair99 There's nothing wrong there. The top one is what you have installed, the bottom one is the one in the modules repo.
> 
> The version difference is because the current fingerprints list is at v108, while you have v106 installed. The prints list is updated from the module (by running the props command), not from the Manager. If there had been an update to the module, the top one would have been under the "Updates" section (as you saw previously) instead of "Installed".

Thanks a lot for the clarification @Didgeridoohan! :)
